app/controllers/api/v1/rooms_controller.rb

```diff
@@ -115,21 +119,34 @@ def recordings_processing
       # GET /api/v1/rooms/:friendly_id/access_code.json
       def access_codes
         access_codes = {
-          viewer_access_code: @room.viewer_access_code
+          viewer_access_code: @room.viewer_access_code,
+          moderator_access_code: @room.moderator_access_code
         }
 
         render_json data: access_codes, status: :ok
       end
 
       # PATCH /api/v1/room_settings/:friendly_id/viewer_access_code.json
-      def viewer_access_code
-        @room.update!(viewer_access_code: SecureRandom.alphanumeric(6).downcase)
+      def generate_access_code
+        case params[:role]
```

Comment:
I wouldn't use `role` as the new here as that could be confused with Greenlight's roles

Reply:
Renamed role to bbb_role